docs/plugin-development/index.md

```diff
@@ -324,7 +328,9 @@ Defines a custom loader. Returning `null` defers to other `load` functions (and
 
 If `false` is returned for `moduleSideEffects` and no other module imports anything from this module, then this module will not be included in the bundle even if the module would have side effects. If `true` is returned, Rollup will use its default algorithm to include all statements in the module that have side effects (such as modifying a global or exported variable). If `"no-treeshake"` is returned, treeshaking will be turned off for this module and it will also be included in one of the generated chunks even if it is empty. If `null` is returned or the flag is omitted, then `moduleSideEffects` will be determined by the first `resolveId` hook that resolved this module, the [`treeshake.moduleSideEffects`](../configuration-options/index.md#treeshake-modulesideeffects) option, or eventually default to `true`. The `transform` hook can override this.
 
-`attributes` contain the import attributes that were used when this module was imported. At the moment, they do not influence rendering for bundled modules but rather serve documentation purposes. If `null` is returned or the flag is omitted, then `attributes` will be determined by the first `resolveId` hook that resolved this module, or the attributes present in the first import of this module. The `transform` hook can override this.
+`options.attributes` contain the import attributes that were used when this module was imported, which is determined by the first `resolveId` hook that resolved this module, or the attributes present in the first import of this module.
+
+The mean of `attributes` is same as `options.attributes`. At the moment, they do not influence rendering for bundled modules but rather serve documentation purposes. If `null` is returned or the flag is omitted, then `attributes` will be equal to `options.attributes`. The `transform` hook can override this.
 
```

Comment:
I keep thinking about this. I find this section slightly confusing, and of course it is: Why would I be able to return attributes unless I want to change them?
As we actually want to remove this functionality soon, I would suggest to actually remove this last part, and also remove `attributes` above from the `SourceDescription` type.

While Rollup will still accept `attributes` at the moment and use them to actually change attributes, we do not want people to use this feature, so there is no need to document it IMO.

Reply:
Yes, I was also very conflicted about how to write the documentation for this content.

Reply:
Thank you for your patient review 😊.